java-api-core/src/com/kinvey/java/query/AbstractQuery.java

```diff
@@ -97,12 +98,16 @@ public String getQueryFilterJson(JsonFactory factory) {
         String jsonResult = "";
         try {
             JsonGenerator generator = factory.createJsonGenerator(writer);
-            generator.serialize(getQueryFilterMap());
+            AbstractMap<String, Object> filterMap = getQueryFilterMap();
+            buildQueryString(generator, filterMap);
             generator.flush();
             jsonResult = writer.toString();
-        } catch (Exception ex) {}
+        } catch (Exception ex) {
+            // TODO add exception handling here instead of suppressing exception
```

Comment:
What's the resolution of this `TODO`? Should we at least print the exception here?

Reply:
Yes, I will throw an exception here.

Reply:
:+1: